lib/types/inet-address.js

```diff
@@ -10,238 +11,243 @@ const utils = require("../utils");
+                "Only IPv4-Mapped IPv6 addresses are allowed as IPv6 address with embedded IPv4 address",
+            );
+        }
+        return new InetAddress(buffer);
     }
-    if (embeddedIp4 && !isValidIPv4Mapped(buffer)) {
-        throw new TypeError(
-            "Only IPv4-Mapped IPv6 addresses are allowed as IPv6 address with embedded IPv4 address",
+
+    /**
+     * Compares 2 addresses and returns true if the underlying bytes are the same
+     * @param {InetAddress} other
+     * @returns {Boolean}
+     */
+    equals(other) {
+        if (!(other instanceof InetAddress)) {
+            return false;
+        }
+        return (
+            this.buffer.length === other.buffer.length &&
+            this.buffer.toString("hex") === other.buffer.toString("hex")
         );
     }
-    return new InetAddress(buffer);
-};
 
-/**
- * Compares 2 addresses and returns true if the underlying bytes are the same
- * @param {InetAddress} other
- * @returns {Boolean}
- */
-InetAddress.prototype.equals = function (other) {
-    if (!(other instanceof InetAddress)) {
-        return false;
+    /**
+     * Returns the underlying buffer
+     * @returns {Buffer}
+     */
+    getBuffer() {
+        return this.buffer;
     }
-    return (
-        this.buffer.length === other.buffer.length &&
-        this.buffer.toString("hex") === other.buffer.toString("hex")
-    );
-};
-
-/**
- * Returns the underlying buffer
- * @returns {Buffer}
- */
-InetAddress.prototype.getBuffer = function () {
-    return this.buffer;
-};
 
-/**
- * Provide the name of the constructor and the string representation
- * @returns {string}
- */
-InetAddress.prototype.inspect = function () {
-    return this.constructor.name + ": " + this.toString();
-};
-
-/**
- * Returns the string representation of the IP address.
- * <p>For v4 IP addresses, a string in the form of d.d.d.d is returned.</p>
- * <p>
- *   For v6 IP addresses, a string in the form of x:x:x:x:x:x:x:x is returned, where the 'x's are the hexadecimal
- *   values of the eight 16-bit pieces of the address, according to rfc5952.
- *   In cases where there is more than one field of only zeros, it can be shortened. For example, 2001:0db8:0:0:0:1:0:1
- *   will be expressed as 2001:0db8::1:0:1.
- * </p>
- * @param {String} [encoding]
- * @returns {String}
- */
-InetAddress.prototype.toString = function (encoding) {
-    if (encoding === "hex") {
-        //backward compatibility: behave in the same way as the buffer
-        return this.buffer.toString("hex");
-    }
-    if (this.buffer.length === 4) {
-        return (
-            this.buffer[0] +
-            "." +
-            this.buffer[1] +
-            "." +
-            this.buffer[2] +
-            "." +
-            this.buffer[3]
-        );
+    /**
+     * Provide the name of the constructor and the string representation
+     * @returns {string}
+     */
+    inspect() {
+        return this.constructor.name + ": " + this.toString();
```

Comment:
```js
`${this.constructor.name}: ${this.toString()}`
```
instead of 
```js
this.constructor.name + ": " + this.toString()
```
It may be personal preference, but maybe consider switching to this way of creating strings?
@wprzytula @ZuzaOsa @Stapox35 @PiotrJunior

Reply:
Definitely, let's use formatting syntax instead of concatenation.